lib/hsa/CMakeLists.txt

```diff
@@ -1,9 +1,13 @@
+include(FindPkgConfig)
+
 ####################
 # HCC runtime (HSA implementation)
 ####################
 if (HAS_ROCM EQUAL 1)
 add_mcwamp_library_hsa(mcwamp_hsa mcwamp_hsa.cpp unpinned_copy_engine.cpp)
-target_link_libraries(mcwamp_hsa  unwind)
+pkg_check_modules(LIBUNWIND REQUIRED libunwind)
+# TODO: we should probably add required CFLAGS and LDFLAGS from pkgconfig
+target_link_libraries(mcwamp_hsa ${LIBUNWIND_LIBRARIES})
```

Comment:
> Can I leave the patch as is, and just add another check to hcc-config.cmake.in?

Yes, but you need to create an imported target, and link against it, like what is done for the hsa libraries. Something like:

```cmake
add_library(unwind INTERFACE IMPORTED)

set_target_properties(unwind PROPERTIES
  INTERFACE_INCLUDE_DIRECTORIES "${LIBUNWIND_INCLUDE_DIRS}"
  INTERFACE_LINK_LIBRARIES "${LIBUNWIND_LIBRARIES}"
  INTERFACE_COMPILE_OPTIONS "${LIBUNWIND_CFLAGS_OTHER}"
  INTERFACE_SYSTEM_INCLUDE_DIRECTORIES "${LIBUNWIND_INCLUDE_DIRS}"
)
target_link_libraries(mcwamp_hsa unwind)
```

And then the imported target needs to be create in the hcc-config.cmake.in file as well.